planner/core/testdata/integration_serial_suite_out.json

```diff
@@ -1102,6 +1102,250 @@
+          "└─ExchangeSender_23 8000.00 cop[tiflash]  ExchangeType: PassThrough",
+          "  └─HashAgg_20 8000.00 cop[tiflash]  group by:test.t.id, funcs:count(Column#7)->Column#4, funcs:firstrow(test.t.id)->test.t.id",
+          "    └─ExchangeReceiver_22 8000.00 cop[tiflash]  ",
+          "      └─ExchangeSender_21 8000.00 cop[tiflash]  ExchangeType: HashPartition, Hash Cols: test.t.id",
+          "        └─HashAgg_8 8000.00 cop[tiflash]  group by:test.t.id, funcs:count(1)->Column#7",
+          "          └─TableFullScan_19 10000.00 cop[tiflash] table:t keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "desc select * from t join ( select count(*), id from t group by id) as A on A.id = t.id",
+        "Plan": [
+          "TableReader_33 9990.00 root  data:ExchangeSender_32",
+          "└─ExchangeSender_32 9990.00 cop[tiflash]  ExchangeType: PassThrough",
+          "  └─HashJoin_9 9990.00 cop[tiflash]  inner join, equal:[eq(test.t.id, test.t.id)]",
+          "    ├─HashAgg_26(Build) 7992.00 cop[tiflash]  group by:test.t.id, funcs:count(Column#8)->Column#7, funcs:firstrow(test.t.id)->test.t.id",
+          "    │ └─ExchangeReceiver_28 7992.00 cop[tiflash]  ",
+          "    │   └─ExchangeSender_27 7992.00 cop[tiflash]  ExchangeType: HashPartition, Hash Cols: test.t.id",
+          "    │     └─HashAgg_17 7992.00 cop[tiflash]  group by:test.t.id, funcs:count(1)->Column#8",
+          "    │       └─Selection_25 9990.00 cop[tiflash]  not(isnull(test.t.id))",
+          "    │         └─TableFullScan_24 10000.00 cop[tiflash] table:t keep order:false, stats:pseudo",
+          "    └─ExchangeReceiver_15(Probe) 9990.00 cop[tiflash]  ",
+          "      └─ExchangeSender_14 9990.00 cop[tiflash]  ExchangeType: HashPartition, Hash Cols: test.t.id",
+          "        └─Selection_13 9990.00 cop[tiflash]  not(isnull(test.t.id))",
+          "          └─TableFullScan_12 10000.00 cop[tiflash] table:t keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "desc select * from t join ( select /*+ hash_agg()*/  count(*) as a from t) as A on A.a = t.id",
+        "Plan": [
+          "HashJoin_10 1.25 root  inner join, equal:[eq(test.t.id, Column#7)]",
+          "├─HashAgg_39(Build) 1.00 root  funcs:count(Column#11)->Column#7",
+          "│ └─TableReader_41 1.00 root  data:ExchangeSender_40",
+          "│   └─ExchangeSender_40 1.00 cop[tiflash]  ExchangeType: PassThrough",
+          "│     └─HashAgg_33 1.00 cop[tiflash]  funcs:count(1)->Column#11",
+          "│       └─TableFullScan_18 10000.00 cop[tiflash] table:t keep order:false, stats:pseudo",
+          "└─TableReader_30(Probe) 9990.00 root  data:Selection_29",
+          "  └─Selection_29 9990.00 cop[tiflash]  not(isnull(test.t.id))",
+          "    └─TableFullScan_28 10000.00 cop[tiflash] table:t keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "desc select avg(value) as b,id from t group by id",
```

Comment:
I find this case push the avg totally, which might change the default behavior. Could you please check out if this will change the plan of TPC-H query 1 and 6 and if this will reduce or speed up the original tpch queries